pkg/trustedresources/verify.go

```diff
@@ -181,43 +277,61 @@ func getVerifiers(ctx context.Context, k8s kubernetes.Interface) ([]signature.Ve
 func verifierForKeyRef(ctx context.Context, keyRef string, hashAlgorithm crypto.Hash, k8s kubernetes.Interface) (verifiers []signature.Verifier, err error) {
 	var raw []byte
 	verifiers = []signature.Verifier{}
-	// if the ref is secret then we fetch the keys from the secrets
-	if strings.HasPrefix(keyRef, keyReference) {
-		s, err := getKeyPairSecret(ctx, keyRef, k8s)
+	v, err := verifierFromSecret(ctx, keyRef, hashAlgorithm, k8s)
+	if err == nil {
+		verifiers = append(verifiers, v)
+	}
+
+	raw, _ = os.ReadFile(filepath.Clean(keyRef))
+	v, err = verifierFromKeyData(raw, hashAlgorithm)
+	if err == nil {
```

Comment:
it seems like we're ignoring the error here, i think instead of adding only when the error it nil, i think we should check for a non-nil error and return an error in that case?